signalfx/provider.go

```diff
@@ -153,27 +172,38 @@ func signalfxConfigure(data *schema.ResourceData) (interface{}, error) {
 		config.CustomAppURL = customAppURL.(string)
 	}
 
-	var netTransport = logging.NewTransport("SignalFx", &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		DialContext: (&net.Dialer{
-			Timeout: 5 * time.Second,
-		}).DialContext,
-		TLSHandshakeTimeout: 5 * time.Second,
-		MaxIdleConns:        100,
-		MaxIdleConnsPerHost: 100,
-	})
+	pooledTransport := cleanhttp.DefaultPooledTransport()
```

Comment:
Is there a reason for using `cleanhttp`? I don't think we modify the transport object elsewhere in the provider, making the risk for race conditions nonexistent.

I don't have an issue using this package if we're trying to make the provider more resilient and there is a chance that we need it in the future; my only concern is keeping up with another package.

Reply:
No reason, unfamiliar with go and ended up there thinking it was needed to get my changes to compile.  I agree with your thinking and removed cleanhttp 👍